### PR TITLE
feat(platform): update client go to 1.24

### PR DESCRIPTION
**Pull request by leoryu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/application/controller/app/app_controller.go [hunk 1]
```diff
@@ -395,7 +403,7 @@ func (c *Controller) syncAppFromRelease(ctx context.Context, cachedApp *cachedAp
 	rel, found := helmutil.Filter(rels, app.Spec.TargetNamespace, app.Spec.Name)
 	if !found {
 		// release not found, reinstall for reconcile
-		newStatus.Phase = applicationv1.AppPhaseInstalling
```
Comment (leoryu) [inline pkg/application/controller/app/app_controller.go:398]: 恢复原样

File: pkg/application/controller/app/app_controller.go [hunk 2]
```diff
@@ -377,6 +382,9 @@ func (c *Controller) handlePhase(ctx context.Context, key string, cachedApp *cac
 }
 
 func (c *Controller) syncAppFromRelease(ctx context.Context, cachedApp *cachedApp, app *applicationv1.App) (*applicationv1.App, error) {
+	if app.Status.Phase == applicationv1.AppPhaseSucceeded && hasSynced(app) {
```
Comment (leoryu) [inline pkg/application/controller/app/app_controller.go:385]: 恢复原样

File: pkg/application/helm/action/client.go [hunk 3]
```diff
@@ -40,7 +41,12 @@ func NewClient(helmDriver string, restClientGetter *config.RESTClientGetter) *Cl
 
 func (c *Client) buildActionConfig(namespace string) (*action.Configuration, error) {
 	actionConfig := new(action.Configuration)
-	if err := actionConfig.Init(c.restClientGetter, namespace, c.helmDriver, log.Debugf); err != nil {
+	err := actionConfig.Init(c.restClientGetter, namespace, c.helmDriver, log.Debugf)
```
Comment (leoryu) [inline pkg/application/helm/action/client.go:44]: @GaoXiaodong check

File: pkg/application/helm/action/install.go [hunk 4]
```diff
@@ -98,6 +99,31 @@ func (c *Client) InstallWithLocal(ctx context.Context, options *InstallOptions,
...
+	rels, err := histClient.Run(options.ReleaseName)
+	if err != nil {
+		if !strings.Contains(err.Error(), "release: not found") {
+			return nil, err
+		}
+	} else {
+		for _, rel := range rels {
+			if rel.Info.Status == release.StatusDeployed {
+				// release 记录已存在，状态为deployed，不再进行重复安装
+				log.Infof("Release %s is already exist. igonre it now.", options.ReleaseName)
```
Comment (leoryu) [inline pkg/application/helm/action/install.go:114]: 恢复原样

File: pkg/platform/util/workqueue/priority_queue.go [hunk 5]
```diff
@@ -229,6 +235,63 @@ func (q *Type) Done(item interface{}) {
...
+		q.waitForProcessing()
+	}
+}
+
+// isProcessing indicates if there are still items on the work queue being
+// processed. It's used to drain the work queue on an eventual shutdown.
+func (q *Type) isProcessing() bool {
+	q.cond.L.Lock()
+	defer q.cond.L.Unlock()
+	return q.processing.len() != 0
```
Comment (leoryu) [inline pkg/platform/util/workqueue/priority_queue.go:264]: @277631272

Comment (leoryu) [reply pkg/platform/util/workqueue/priority_queue.go:264]: @277631272

**Review by willzgli — CHANGES_REQUESTED** [review]

File: pkg/application/helm/action/client.go (hunk 3 above)
Comment (GaoXiaodong) [reply pkg/application/helm/action/client.go:44]: 必须给RegistryClient赋值，否则会调用helm方法的时候panic

Comment (GaoXiaodong) [reply pkg/application/helm/action/client.go:44]: 确认这里没问题

File: pkg/application/controller/app/action/install.go [hunk 6]
```diff
@@ -71,60 +89,110 @@ func Install(ctx context.Context,
...
+		chartPathBasicOptions, err := chartpath.BuildChartPathBasicOptions(repo, newApp.Spec.Chart)
+		if err != nil {
+			return nil, err
+		}
+		chartPathBasicOptions.ExistedFile = destfile
 
-	chartPathBasicOptions, err := chartpath.BuildChartPathBasicOptions(repo, newApp.Spec.Chart)
-	if err != nil {
-		return nil, err
+		/* provide compatibility with online tke addon apps */
```
[the remark this answers is not in the document]
Comment (GaoXiaodong) [reply pkg/application/controller/app/action/install.go:107]: /* provide compatibility with online tke addon apps */
/* compatibility over, above code need to be deleted atfer the online addon apps are migrated */
这段注释中间的不要了，后面我们升级的时候刷下存量，辛苦 @leoryu 帮处理下

File: pkg/application/controller/app/action/upgrade.go [hunk 7]
```diff
@@ -73,36 +88,88 @@ func Upgrade(ctx context.Context,
...
+		values, err := helmutil.MergeValues(app.Spec.Values.Values, app.Spec.Values.RawValues, string(app.Spec.Values.RawValuesType))
+		if err != nil {
+			return nil, err
+		}
+		chartPathBasicOptions, err := chartpath.BuildChartPathBasicOptions(repo, newApp.Spec.Chart)
+		if err != nil {
+			return nil, err
+		}
+		chartPathBasicOptions.ExistedFile = destfile
+		//* provide compatibility with online tke addon apps */
```
Comment (GaoXiaodong) [inline pkg/application/controller/app/action/upgrade.go:105]: //* provide compatibility with online tke addon apps */
//* compatibility over, above code need to be deleted atfer the online addon apps are migrated */
这两个注释间的也不要了，辛苦 @leoryu 帮处理下

**Pull request merged**